File: frontend/src/types/personProject.ts
import type { PersonProjectStatus, FailReason } from '.'

export interface PersonProject {
  id: number
  person_id: number
  project_id: number
  owner_id: number
  status: PersonProjectStatus
  fail_reason: FailReason | null
  fail_remark: string | null
  attended_training: boolean | null
  purchased_package: boolean | null
  completed_rooms: number | null
  created_at: string
  updated_at: string
  // 关联数据
  person_name?: string
  person_phone?: string
  project_name?: string
  enterprise_name?: string
  owner_name?: string
}

export interface PersonProjectCreate {
  person_id: number
  project_id: number
  owner_id: number
}

export interface StatusChangeRequest {
  status: PersonProjectStatus
  fail_reason?: FailReason
  fail_remark?: string
}

export interface BatchStatusChangeRequest {
  person_project_ids: number[]
  status: PersonProjectStatus
  fail_reason?: FailReason
  fail_remark?: string
}

export interface ReassignRequest {
  project_id: number
  owner_id: number
}

File: backend/app/schemas/person_project.py
"""人员-项目关联 schemas。"""

from datetime import datetime

from pydantic import BaseModel

from app.models.enums import FailReason, PersonProjectStatus


class PersonProjectBase(BaseModel):
    """人员项目关联基础模型。"""

    person_id: int
    project_id: int
    owner_id: int


class PersonProjectCreate(PersonProjectBase):
    """创建人员项目关联（分配人员到项目）。"""

    pass


class StatusChangeRequest(BaseModel):
    """状态变更请求。"""

    status: PersonProjectStatus
    fail_reason: FailReason | None = None
    fail_remark: str | None = None


# 兼容旧字段名
StatusAdvanceRequest = StatusChangeRequest


class PersonProjectResponse(PersonProjectBase):
    """人员项目关联响应。"""

    id: int
    status: PersonProjectStatus
    fail_reason: FailReason | None
    fail_remark: str | None
    attended_training: bool | None
    purchased_package: bool | None
    completed_rooms: int | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PersonProjectDetailResponse(PersonProjectResponse):
    """人员项目关联详情响应。"""

    person_name: str
    person_phone: str
    project_name: str
    enterprise_name: str
    owner_name: str


class ReassignRequest(BaseModel):
    """人力池再分配请求。"""

    project_id: int
    owner_id: int


class ReassignResponse(BaseModel):
    """人力池再分配响应。"""

    person_project_id: int
    person_id: int
    project_id: int
    status: PersonProjectStatus
